src/main/java/org/terasology/module/behaviors/actions/FindPathToNode.java

```diff
@@ -2,95 +2,87 @@
-        pathfinderSystem = CoreRegistry.get(PathfinderSystem.class);
-    }
 
     @Override
-    public void construct(final Actor actor) {
+    public void construct(Actor actor) {
         if (pathfinderSystem == null) {
-            setup();
+            pathfinderSystem = CoreRegistry.get(PathfinderSystem.class);
         }
-        final MinionMoveComponent moveComponent = actor.getComponent(MinionMoveComponent.class);
-        Vector3f targetLocation = moveComponent.target;
-        moveComponent.path = null;
-        actor.save(moveComponent);
-        WalkableBlock currentBlock = moveComponent.currentBlock;
-        if (currentBlock == null || targetLocation == null) {
-            moveComponent.path = Path.INVALID;
-            return;
+        if (pluginSystem == null) {
+            pluginSystem = CoreRegistry.get(PluginSystem.class);
         }
-        WalkableBlock workTarget = navGraphSystem.getBlock(targetLocation);
-        if (workTarget == null) {
-            moveComponent.path = Path.INVALID;
-            return;
-        }
-        SettableFuture<List<Path>> pathFuture = pathfinderSystem.requestPath(
-                actor.getEntity(), currentBlock.getBlockPosition(),
-                Arrays.asList(workTarget.getBlockPosition()));
 
-        Futures.addCallback(pathFuture, new FutureCallback<List<Path>>() {
-            @Override
-            public void onSuccess(List<Path> paths) {
-                if (paths == null) {
-                    moveComponent.path = Path.INVALID;
-                } else if (paths.size() > 0) {
-                    moveComponent.path = paths.get(0);
-                }
-                actor.save(moveComponent);
-            }
+        MinionMoveComponent flexibleMovementComponent = actor.getComponent(MinionMoveComponent.class);
+        flexibleMovementComponent.running = true;
+        Vector3ic start = Blocks.toBlockPos(actor.getComponent(LocationComponent.class).getWorldPosition(new Vector3f()));
+        Vector3ic goal = actor.getComponent(MinionMoveComponent.class).getPathGoal();
 
-            @Override
-            public void onFailure(Throwable t) {
-                moveComponent.path = Path.INVALID;
+        JPSConfig config = new JPSConfig(start, goal);
+        config.useLineOfSight = false;
+        config.requester = actor.getEntity();
+        config.maxTime = 10f;
+        config.maxDepth = 150;
+        config.goalDistance = flexibleMovementComponent.pathGoalDistance;
+        config.plugin = pluginSystem.getMovementPlugin(actor.getEntity()).getJpsPlugin(actor.getEntity());
+
+        int id = pathfinderSystem.requestPath(config, (path, target) -> {
+            MinionMoveComponent flexibleMovementComponent1 = actor.getComponent(MinionMoveComponent.class);
+            flexibleMovementComponent1.running = false;
+            if (path == null || path.size() == 0) {
+                actor.save(flexibleMovementComponent1);
+                return;
```

Comment:
This has the changes around `MinionMoveComponent#running` which were added in an attempt to understand what's happening and to fix potential issues. It was not there in the original code from FlexibleMovement, so maybe we should revert [`d136af9` (#89)](https://github.com/Terasology/Behaviors/pull/89/commits/d136af90cdb516e2f66ac551a5834dc2019353bc).